src/OpenDebugAD7/VariableManager.cs

```diff
@@ -88,6 +88,11 @@ internal Variable CreateVariable(ref DEBUG_PROPERTY_INFO propertyInfo, enum_DEBU
                 EvaluateName = propertyInfo.bstrFullName,
                 MemoryReference = memoryReference
             };
+
+            if (propertyInfo.dwAttrib.HasFlag(enum_DBG_ATTRIB_FLAGS.DBG_ATTRIB_VALUE_READONLY))
+                v.PresentationHint = new VariablePresentationHint() { Attributes = VariablePresentationHint.AttributesValue.ReadOnly };
```

Comment:
The thing is it does get sent but VSCode ignores it.
```json
{"name":"rax","value":"0x0","presentationHint":{"attributes":["readOnly"]}
```

Reply:
It's also interesting that this check doesn't trigger.
https://github.com/microsoft/MIEngine/blob/3c480786f26ec5340ff16dfeb8d7fd8cde077a9d/src/OpenDebugAD7/AD7DebugSession.cs#L1707-L1713

Instead I get `Stopping due to fatal error: NullReferenceException: Object reference not set to an instance of an object
`.

Reply:
> The thing is it does get sent but VSCode ignores it.

Yes, this was more to make VS happy.

> Instead I get Stopping due to fatal error: NullReferenceException: Object reference not set to an instance of an object .

The issue is that [this code](https://github.com/Microsoft/MIEngine/blob/896c72d633b39a0571b9599d15c5a8fd0f1fcf58/src/MIDebugEngine/AD7.Impl/AD7RegProperty.cs#L59-L79) doesn't provide a 'pProperty'. The easy fix is to delete line 1702 and 1703, and change the rest of the code in that block to use `props` instead of `propertyInfo` (or just rename `props`).

Reply:
Doing that brings us to the `Error_VariableNotFound` part.
I guess because registers are grouped ("CPU", "SSE", "AVX", ...) and without 1702-3 it looks in the wrong level. But all that property code is Greek to me in general.

Reply:
It shouldn't change the 'VariableNotFound' code. What is going on is that for some reason this code was calling `GetPropertyInfo` when it should already have a property info from calling `Next`. So this just removes the refetch of the same data.

Reply:
I meant that it reports VariableNotFound instead of variable is read-only.

Reply:
The issue is `props` has no data so it can not match the register that was sent from the `HandleSetVariableRequestAsync`.
![image](https://user-images.githubusercontent.com/3953714/105253230-efd36380-5b33-11eb-95c1-34d71dea5998.png)

I believe this needs to also support the `readOnly` attribute for variables.